app.js

```diff
@@ -1,15 +1,30 @@
 const express = require('express');
 const app = express();
-const { getTopics, getAllEndpoints } = require('./controlers/controlers');
+const {
+  getTopics,
+  getAllEndpoints,
+  getArticleById,
+} = require('./controlers/controlers');
 
 app.use(express.json());
 
 app.get('/api/topics', getTopics);
 app.get('/api', getAllEndpoints);
-
+app.get('/api/articles/:article_id', getArticleById);
+app.use((err, req, res, next) => {
+  if (err.code === '22P02') {
+    res.status(400).send({ msg: 'Bad request' });
+  } else {
+    next(err);
+  }
+});
 app.use((err, req, res, next) => {
   console.log(err);
-  next(err);
+  if (err.msg === 'article does not exist') {
+    res.status(404).send({ msg: err.msg });
+  } else {
+    next(err);
+  }
```

Comment:
Good work here also. Currently it's setup to execute only if the `.msg` key is `article does not exist` - think about how you can extend the functionality of this, maybe check if the `err` object has a `msg` and `status` property and if it does... send the corresponding status code and msg. There's some nice info our notes here about it: https://notes.northcoders.com/courses/js-back-end/error-handling#custom-errors